src/main/frontend/handler/link.cljs

```diff
@@ -1,28 +1,38 @@
 (ns frontend.handler.link
   (:require [frontend.util :as util]))
 
-(def plain-link "(\bhttps?://)?[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]")
-(def org-link-re-1 (re-pattern (util/format "\\[\\[(%s)\\]\\[(.+)\\]\\]" plain-link)))
-(def org-link-re-2 (re-pattern (util/format "\\[\\[(%s)\\]\\]" plain-link)))
-(def markdown-link-re (re-pattern (util/format "^\\[(.+)\\]\\((%s)\\)" plain-link)))
+(def plain-link-re-string
+  "(?:http://www\\.|https://www\\.|http://|https://){1}[a-z0-9]+(?:[\\-\\.]{1}[a-z0-9]+)*\\.[a-z]{2,5}(?:.*)*")
+;; Based on https://orgmode.org/manual/Link-Format.html#Link-Format
+(def org-link-re-1 (re-pattern (util/format "\\[\\[(%s)\\]\\[(.+)\\]\\]" plain-link-re-string)))
+(def org-link-re-2 (re-pattern (util/format "\\[\\[(%s)\\]\\]" plain-link-re-string)))
+(def markdown-link-re (re-pattern (util/format "^\\[(.+)\\]\\((%s)\\)" plain-link-re-string)))
 
-(defn- org-link?
+(defn- org-link
   [link]
   (when-let [matches (or (re-matches org-link-re-1 link)
                          (re-matches org-link-re-2 link))]
     {:type "org-link"
      :url (second matches)
      :label (nth matches 2 nil)}))
 
-(defn- markdown-link?
+(defn- markdown-link
   [link]
   (when-let [matches (re-matches markdown-link-re link)]
     {:type "markdown-link"
      :url (nth matches 2)
      :label (second matches)}))
 
-(defn link?
+(defn- plain-link
   [link]
-  (or (util/url? link)
-      (org-link? link)
-      (markdown-link? link)))
+  (when (util/url? link)
+    {:type "plain-link"
+     :url link}))
+
+(defn link
```

Comment:
I renamed this fn to `link` as it caused a slight bug in https://github.com/logseq/logseq/commit/2401ce9e9ab6ce3e7d2ab63ab4a1a7e49ec6f5ee#diff-d5720cb6221452c421a80ce9e8df86cce229e69ac53b10506161ffcfbf3d6889 where a boolean was returned instead of a map with `util/url?`, probably because of the name confusion. Fns ending in `?` normally return boolean values, which this fn doesn't